internal/jet/array_expression.go

```diff
@@ -0,0 +1,93 @@
+package jet
+
+// ArrayExpression interface
+type ArrayExpression[E Expression] interface {
```

Comment:
Maybe just `Array[E Expression]`, so it follows naming convention with `Range`. Also `Expression` would repeat twice in the name.

Reply:
👍 Done